docs/kb/edk2-capsule-updates.md

```diff
@@ -325,6 +328,326 @@ interferes).
 
 [wiki-pkc]: https://en.wikipedia.org/wiki/Public-key_cryptography
 
+## Generating signing keys with OpenSSL
+
+The process involves creation of 3 certificates, a local certificate
+authority (CA) in a directory and 14 files.  This warrants some overview of
+what's going to be done.
+
+Keys making up the chain will assume the following roles:
+
+- _root_ — self-signed certificate acting as the basis of a CA
+- _sub_ — second-level CA signed by _root_ key
+- _sign_ — a signing key (not a CA) that is signed by _sub_ key
+
+A digital signature algorithm (DSA) consists of a private key (PK) algorithm and
+a digest algorithm.  Keys in a chain don't have to use the same DSA.  Also, most
+DSAs allow use of an arbitrary digest (not Ed25519 or Ed448 which prescribe the
+use of SHA-512 and SHAKE-256 respectively).
+
+Keys exist independently of the chains in which they appear.  When stored,
+private keys are often encrypted.
+
+### Key generation by example
+
+It will be easier to go into finer details while looking at the actual commands.
+The entire process can be broken into several stages:
+
+1. Creation of 3 certificates
+2. Construction of a CA
+3. Preparation of _sign_ certificate for signing
+4. Preparation of _root_ certificate for inclusion into EDK
```

Comment:
I think those steps should be swapped, to show how they are used chronologically in build process. Same for their descriptions below.

Reply:
https://github.com/Dasharo/docs/compare/c791ff82210eed388fe9a65c78a80490c8d13b24..fcda915b6e88ce72089bfc5a43900ac61336c7ba#diff-b0d92435e1999e660dd9b88bd5bd5dd14331f32b3abce08c0b9a5895537af576